batch_consumer.go

```diff
@@ -86,29 +93,35 @@ func (b *batchConsumer) startBatch() {
 	defer ticker.Stop()
 
 	maximumMessageLimit := b.messageGroupLimit * b.concurrency
+	maximumMessageByteSizeLimit := b.messageGroupByteSizeLimit * b.concurrency
 	messages := make([]*Message, 0, maximumMessageLimit)
 	commitMessages := make([]kafka.Message, 0, maximumMessageLimit)
-
+	messageByteSize := 0
 	for {
 		select {
 		case <-ticker.C:
 			if len(messages) == 0 {
 				continue
 			}
 
-			b.consume(&messages, &commitMessages)
+			b.consume(&messages, &commitMessages, &messageByteSize)
 		case msg, ok := <-b.incomingMessageStream:
 			if !ok {
 				close(b.batchConsumingStream)
 				close(b.messageProcessedStream)
 				return
 			}
 
+			if maximumMessageByteSizeLimit != 0 && messageByteSize+len(msg.message.Value) > maximumMessageByteSizeLimit {
```

Comment:
Instead of 
```go
messageByteSize+len(msg.message.Value)
```

we could do same as segmentio/kafka-go does
```go
	for i := range msgs {
		n := int64(msgs[i].totalSize())
		if n > batchBytes {
			...
		}
	}

        func (msg *Message) totalSize() int32 {
	        return int32(msg.headerSize()) + msg.size()
        }
       func (msg *Message) size() int32 {
	         return 4 + 1 + 1 + sizeofBytes(msg.Key) + sizeofBytes(msg.Value) + timestampSize
       }

        func (msg *Message) headerSize() int {
	         return varArrayLen(len(msg.Headers), func(i int) int {
		     h := &msg.Headers[i]
                     return varStringLen(h.Key) + varBytesLen(h.Value)
	       })
}
```
etc.